modules/dnn/src/dnn_utils.cpp

```diff
@@ -102,11 +102,18 @@ void blobFromImagesWithParams(InputArrayOfArrays images_, OutputArray blob_, con
 
     int nch = images[0].channels();
     Scalar scalefactor = param.scalefactor;
+    Scalar mean = param.mean;
 
     if (param.ddepth == CV_8U)
     {
         CV_Assert(scalefactor == Scalar::all(1.0) && "Scaling is not supported for CV_8U blob depth");
-        CV_Assert(param.mean == Scalar() && "Mean subtraction is not supported for CV_8U blob depth");
+        CV_Assert(mean == Scalar() && "Mean subtraction is not supported for CV_8U blob depth");
+    }
+
+    if (nch > 2 && param.swapRB)
+    {
+        std::swap(mean[0], mean[2]);
+        std::swap(scalefactor[0], scalefactor[2]);
     }
```

Comment:
I propose to add CV_LOG_WARNING for the else case to highlight the issue.

Reply:
What do you think about that?
```
if (param.swapRB)
{
    if (nch > 2)
    {
        std::swap(mean[0], mean[2]);
        std::swap(scalefactor[0], scalefactor[2]);
    }
    else
    {
        CV_LOG_WARNING(NULL, "Red/blue color swapping requires at least three image channels.");
    }
}
```